### PR TITLE
[ASTGen] Allow querying of experimental features

### PR DESCRIPTION
**Pull request by hamishknight**
Update for https://github.com/apple/swift-syntax/pull/2041 such that experimental features can be enabled in SwiftSyntax.

### CODE REVIEW

**Comment by hamishknight** [comment]
https://github.com/apple/swift-syntax/pull/2041

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/swift-syntax/pull/2041

@swift-ci please test

File: lib/ASTGen/Sources/ASTGen/SourceFile.swift [hunk 1]
```diff
@@ -20,15 +25,31 @@ struct ExportedSourceFile {
...
+extension Parser.ExperimentalFeatures {
+  init(from context: BridgedASTContext?) {
+    self = []
+    guard let context = context else { return }
+
+    func mapFeature(_ bridged: BridgedFeature, to feature: Self) {
+      if ASTContext_langOptsHasFeature(context, bridged) {
+        insert(feature)
+      }
+    }
```
Comment (ahoppen) [inline lib/ASTGen/Sources/ASTGen/SourceFile.swift:37]: I assume this is a stub to be used once you introduce features that need to be mapped to SwiftParser, correct?

**Review by ahoppen — APPROVED** [review]

File: lib/ASTGen/Sources/ASTGen/SourceFile.swift (hunk 1 above)
Comment (hamishknight) [reply lib/ASTGen/Sources/ASTGen/SourceFile.swift:37]: Yeah

**Pull request merged**